src/test/java/teammates/test/cases/util/StringHelperTest.java

```diff
@@ -202,6 +202,22 @@ public void testRecoverFromSanitizedText() {
         str = Sanitizer.sanitizeForHtml("<text><div> 'param' &&& \\//\\");
         assertEquals("<text><div> 'param' &&& \\//\\", StringHelper.recoverFromSanitizedText(str));
     }
+    
+    @Test
+    public void testReplaceIllegalChars() {
+        String str = null;
+        assertEquals(null, StringHelper.replaceIllegalChars(str, FieldValidator.REGEX_COURSE_ID, '_'));
```

Comment:
Since this is a unit test, can you hard-code some regex variable here instead of using something from `FieldValidator`? The idea is to reduce unnecessary coupling.
```java
String regex = "..."; // perfectly fine if this is copy-pasted from FieldValidator.REGEX_COURSE_ID
String str = null;
assertEquals(null, ...);
...
```